### PR TITLE
PUB-654 - Added findByUserId API

### PR DESCRIPTION
**Pull request by commitBlob**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PUB-654

### Change description ###
Added the ability to pull a list of subscriptions back related to a specific user.

### API
created api that takes in a user Id and then uses that to find all subscriptions for that user, if it cant find any for the user it returns an empty object (ie. the user doesnt have any subscriptions yet) else it returns a populated object that looks like:
```
caseSubscriptions: [{caseName: "test name", caseNumber: "123", urn: "321", dateAdded: "11/11/11"}],
courtSubscriptions: [{courtName: "court name", dateAdded: "11/11/11"}]
```

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by furnivall — APPROVED** [review]
Looks great 🥇

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/models/response/UserSubscription.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import lombok.Data;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Response object for a users subscriptions that returns both Courts and Hearings that a user is subscribed to.
+ */
+@Data
+public class UserSubscription {
```
[the remark this answers is not in the document]
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/models/response/UserSubscription.java:11]: good spot! ill get that added!

[the remark this answers is not in the document]
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/models/response/UserSubscription.java:12]: its been added to the CaseSubscription and CourtSubscription model as these are the ones that hold individual subscription details 😊

**Review by ChrisS1512 — DISMISSED** [review]

**Review by furnivall — DISMISSED** [review]
Looks good to me :)

**Review by furnivall — DISMISSED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
👍

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**